modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-channel-impl/src/main/java/com/liferay/headless/commerce/admin/channel/internal/resource/v1_0/DefaultCategoryDisplayPageResourceImpl.java

```diff
@@ -0,0 +1,231 @@
+		modifiableSettings.store();
+	}
+
+	public DefaultCategoryDisplayPage
+			getChannelByExternalReferenceCodeDefaultCategoryDisplayPage(
+				String externalReferenceCode)
+		throws Exception {
+
+		CommerceChannel commerceChannel =
+			_commerceChannelLocalService.
+				getCommerceChannelByExternalReferenceCode(
+					externalReferenceCode, contextCompany.getCompanyId());
+
+		return getChannelIdDefaultCategoryDisplayPage(
+			commerceChannel.getCommerceChannelId());
+	}
+
+	public DefaultCategoryDisplayPage getChannelIdDefaultCategoryDisplayPage(
+			Long id)
+		throws Exception {
+
+		CommerceChannel commerceChannel =
+			_commerceChannelLocalService.getCommerceChannel(id);
+
+		Settings settings = FallbackKeysSettingsUtil.getSettings(
+			new GroupServiceSettingsLocator(
+				commerceChannel.getGroupId(),
+				CPConstants.RESOURCE_NAME_CP_DISPLAY_LAYOUT));
+
+		ModifiableSettings modifiableSettings =
+			settings.getModifiableSettings();
+
+		String assetCategoryLayoutUuid = modifiableSettings.getValue(
+			"assetCategoryLayoutUuid", StringPool.BLANK);
+
+		if (Validator.isBlank(assetCategoryLayoutUuid)) {
+			throw new NoSuchCPDisplayLayoutException();
+		}
+
+		return _toDefaultCategoryDisplayPage(commerceChannel);
+	}
+
+	public DefaultCategoryDisplayPage
+			postChannelByExternalReferenceCodeDefaultCategoryDisplayPage(
+				String externalReferenceCode,
+				DefaultCategoryDisplayPage defaultCategoryDisplayPage)
+		throws Exception {
+
+		CommerceChannel commerceChannel =
+			_commerceChannelLocalService.
+				getCommerceChannelByExternalReferenceCode(
+					externalReferenceCode, contextCompany.getCompanyId());
+
+		return postChannelIdDefaultCategoryDisplayPage(
+			commerceChannel.getCommerceChannelId(), defaultCategoryDisplayPage);
+	}
+
+	public DefaultCategoryDisplayPage postChannelIdDefaultCategoryDisplayPage(
+			Long id, DefaultCategoryDisplayPage defaultCategoryDisplayPage)
+		throws Exception {
+
+		CommerceChannel commerceChannel =
+			_commerceChannelLocalService.getCommerceChannel(id);
+
+		_validateLayoutUuid(
+			commerceChannel.getSiteGroupId(),
+			defaultCategoryDisplayPage.getPageUuid());
+
+		Settings settings = FallbackKeysSettingsUtil.getSettings(
+			new GroupServiceSettingsLocator(
+				commerceChannel.getGroupId(),
+				CPConstants.RESOURCE_NAME_CP_DISPLAY_LAYOUT));
+
+		ModifiableSettings modifiableSettings =
+			settings.getModifiableSettings();
+
+		modifiableSettings.setValue(
+			"assetCategoryLayoutUuid",
+			defaultCategoryDisplayPage.getPageUuid());
+
+		modifiableSettings.store();
+
+		return _toDefaultCategoryDisplayPage(commerceChannel);
+	}
+
+	private Map<String, Map<String, String>> _getActions(
+		CommerceChannel commerceChannel) {
+
+		return HashMapBuilder.<String, Map<String, String>>put(
+			"delete",
+			addAction(
+				ActionKeys.DELETE, commerceChannel.getCommerceChannelId(),
+				"deleteChannelIdDefaultCategoryDisplayPage",
+				commerceChannel.getUserId(), CommerceChannel.class.getName(),
```

Comment:
I'm not sure you should use the channel userId, why don't you use the contextUser?